traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java

```diff
@@ -18,17 +14,14 @@
 import org.json.JSONException;
 import org.json.JSONObject;
 
-import java.io.OutputStream;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.util.*;
 
-import static com.guestlogix.travelercorekit.utilities.UrlHelper.urlEncodeUTF8;
-
 
 public class Router {
```

Comment:
I changed Router.java to use a private getSessionToken that handles the edge case of a session's Token being null.
I don't think this will happen in the app, since the lib is initialized early on; but in the instrumented test the API call happens very quickly, before the token is fetched.

.build() with a null token value seems to be fine; the task naturally fetches a new token.

Reply:
Lets talk about this. There shouldnt be a need for this. null should go fine and then refetch the token.

Reply:
The problem is that when `session.getToken()` resolves to null, `.getValue()` causes NPE

Reply:
will change to passing in TOKEN rather than its VALUE as discussed